flink-runtime/src/main/java/org/apache/flink/runtime/registration/RetryingRegistration.java

```diff
@@ -329,19 +329,23 @@ private void register(final G gateway, final int attempt, final long timeoutMill
 
     private void registerLater(
             final G gateway, final int attempt, final long timeoutMillis, long delay) {
-        rpcService.scheduleRunnable(
-                new Runnable() {
-                    @Override
-                    public void run() {
-                        register(gateway, attempt, timeoutMillis);
-                    }
-                },
-                delay,
-                TimeUnit.MILLISECONDS);
+        rpcService
+                .getScheduledExecutor()
+                .schedule(
+                        new Runnable() {
+                            @Override
+                            public void run() {
+                                register(gateway, attempt, timeoutMillis);
+                            }
+                        },
+                        delay,
+                        TimeUnit.MILLISECONDS);
```

Comment:
```suggestion
                .schedule(
                        () -> register(gateway, attempt, timeoutMillis),
                        delay,
                        TimeUnit.MILLISECONDS);
```
nit: what about also beautifying the code a bit as part of this change.